docs/billing-integration.md

```diff
@@ -0,0 +1,273 @@
+      # occur for this metric. Reports received are immediately sent
+      # to the reporting endpoint. We use passthrough for the
+      # instance_time metric since reports are generated
+      # automatically by a heartbeat source defined in a later
+      # section.
+      passthrough: {}
+
+    # The endpoints section defines where metering data is ultimately
+    # sent. Currently supported endpoints include:
+    # * disk - some directory on the local filesystem
+    # * servicecontrol - Google Service Control
+    endpoints:
+    - name: on_disk
+      # The disk endpoint is useful for debugging, but its inclusion
+      # is not necessary in a production deployment.
+      disk:
+        reportDir: /var/lib/ubbagent/reports
+        expireSeconds: 3600
+    - name: servicecontrol
+      servicecontrol:
+        identity: gcp
+        # The service name is unique to your application and will be
+        # provided during onboarding.
+        serviceName: my-application.mp-my-company.appspot.com
+        consumerId: $AGENT_CONSUMER_ID  # From the reporting secret.
+      
+
+    # The sources section lists metric data sources run by the agent
+    # itself. The currently-supported source is 'heartbeat', which
+    # sends a defined value to a metric at a defined interval. In
+    # this example, the heartbeat sends a 60-second value through the
+    # "instance_time" metric every minute.
+    sources:
+    - name: instance_time_heartbeat
+      heartbeat:
+        metric: instance_time
+        intervalSeconds: 60
+        value:
+          int64Value: 60
+---
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: $name-metered-app
+  labels: &AppDeploymentLabels
+    app.kubernetes.io/name: "$name"
+    app.kubernetes.io/component: my-application
+spec:
+  replicas: 1
+  selector:
+    matchLabels: *AppDeploymentLabels
+  template:
+    metadata:
+      labels: *AppDeploymentLabels
+    spec:
+      containers:
+      - name: my-app
+        image: $imageMyApp
+        # ... your app container's config ...
+      - name: ubbagent
+        image: $imageUbbagent
+        env:
+        - name: AGENT_CONFIG_FILE
+          value: "/etc/ubbagent/config.yaml"
+        - name: AGENT_LOCAL_PORT
+          value: "4567"
+        - name: AGENT_ENCODED_KEY
+          valueFrom:
+            secretKeyRef:
+              name: $reportingSecret
+              key: reporting-key
+        - name: AGENT_CONSUMER_ID
+          valueFrom:
+            secretKeyRef:
+              name: $reportingSecret
+              key: consumer-id
+        volumeMounts:
+        - name: ubbagent-config
+          mountPath: /etc/ubbagent
+      volumes:
+      - name: ubbagent-config
+        configMap:
+          name: ubbagent-config
+```
+
+`schema.yaml` excerpt:
+```yaml
+application_api_version: v1beta1
+properties:
+  name:
+    type: string
+    x-google-marketplace:
+      type: NAME
+  imageMyApp:
+    type: string
+    default: $REGISTRY:$TAG
```

Comment:
Replace variables with real values.

These are intended to be literal image URLs. We just happen to use substitutions in our example app for automation, which requires `envsubst` invocation in the Dockerfile.